bashrc

```diff
@@ -756,26 +757,38 @@ fkill() {
 #   - Bypass fuzzy finder if there's only one match (--select-1)
 #   - Exit if there's no match (--exit-0)
 fe() {
-  local files
-  IFS=$'\n' files=($(fzf-tmux --query="$1" --multi --select-1 --exact --exit-0))
-  [[ -n "$files" ]] && ${EDITOR:-vim} "${files[@]}"
+	local files
+	IFS=$'\n' files=($(fzf-tmux --query="$1" --multi --select-1 --exact --exit-0))
+	[[ -n "$files" ]] && ${EDITOR:-vim} "${files[@]}"
 }
 
 # fh - repeat history
 runcmd (){ perl -e 'ioctl STDOUT, 0x5412, $_ for split //, <>' ; }
 
 fh() {
-  ([ -n "$ZSH_NAME" ] && fc -l 1 || history) | fzf +s --tac | sed -re 's/^\s*[0-9]+\s*//' | runcmd
+  ([ -n "$ZSH_NAME" ] && fc -l 1 || history)\
+    | fzf +s --tac | sed -re 's/^\s*[0-9]+\s*//' | runcmd
 }
 
-fcd() {     if [[ "$#" != 0 ]]; then         builtin cd "$@";         return;     fi;     while true; do         local lsd=$(echo ".." && ls -p | grep '/$' | sed 's;/$;;');         local dir="$(printf '%s\n' "${lsd[@]}" |
-            fzf --reverse --preview '
-                __cd_nxt="$(echo {})";
-                __cd_path="$(echo $(pwd)/${__cd_nxt} | sed "s;//;/;")";
-                echo $__cd_path;
-                echo;
-                ls -p --color=always "${__cd_path}";
-        ')";         [[ ${#dir} != 0 ]] || return 0;         builtin cd "$dir" &> /dev/null;     done; }
+fcd() {
+	if [[ "$#" != 0 ]]; then
+		builtin cd "$@"
+		return
+	fi
+
+	while true; do
+		local lsd=$(echo ".." && ls -p | grep '/$' | sed 's;/$;;')
+		local dir="$(printf '%s\n' "${lsd[@]}" | fzf --reverse --preview '
+			__cd_nxt="$(echo {})";
+			__cd_path="$(echo $(pwd)/${__cd_nxt} | sed "s;//;/;")";
+			echo $__cd_path;
+			echo;
+			ls -p --color=always "${__cd_path}";')"
+
+		[[ ${#dir} != 0 ]] || return 0
+		builtin cd "$dir" &> /dev/null
+	done
+}
```

Comment:
This was originally very messy. I believe it's now vastly easier to read and edit.